### PR TITLE
Update Repo structure batch 4

### PR DESCRIPTION
**Pull request by chidozieononiwu**
- Updated service location repo root/sdk
- Updated package structure.
- Renamed projects to start with Microsoft.Azure.Batch
- Updated to project properties to match new engineering system.
- Updated TargetFramework to use $(RequiredTargetFramework) i.e. net452, netstandard1.4, net461 and netcoreapp2.1 for test projects.
- Updated CertificateBuilder.cs to run for netcore using BountyCastle instead of Mono.Security.
- Updated CertificateUnitTests.cs
- Disabled CerficaCredentials return on IntegrationTestCommon.cs
- Used Async methods to accomodate .netcore2.1
- Added [LiveTest] Attribute for test requiring environment variable.
- Removed tests in IntegrationJobApplicationPackageReferencesTests.cs due to wrong parameters in BatchServiceClient

### CODE REVIEW

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTestCommon/CertificateBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (matthchr) [inline sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTestCommon/CertificateBuilder.cs]: Any idea why this file is flagged as a new file and not just a move?

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/ComputeNodeIntegrationTests.cs [hunk 2]
```diff
@@ -829,7 +839,7 @@ public void ComputeNodeUploadLogs()
                         DateTime timeoutAt = DateTime.UtcNow.AddMinutes(2);
                         while (DateTime.UtcNow < timeoutAt)
                         {
-                            blobs = container.ListBlobs();
+                            blobs = await BlobStorageExtensions.ListBlobs(container);
```
Comment (matthchr) [inline sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/ComputeNodeIntegrationTests.cs]: Does this exist because storage doesn't have continuation handling in their netcore SDK?

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/IntegrationTestUtilities/FileUploadUtilities.cs [hunk 3]
```diff
@@ -252,7 +252,7 @@ private static string ConstructBlobSource(string container, string blob)
         /// <param name="permissions">permission on the name</param>
         /// <param name="blobUri">blob URI</param>
         /// <returns>the SAS for the container, in full URI format.</returns>
-        private static string CreateContainerWithPolicySASIfNotExist(string account, string key, Uri blobUri, string container, string policy, DateTime start, DateTime end, SharedAccessBlobPermissions permissions)
+        private static async Task<string> CreateContainerWithPolicySASIfNotExist(string account, string key, Uri blobUri, string container, string policy, DateTime start, DateTime end, SharedAccessBlobPermissions permissions)
```
Comment (matthchr) [inline sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/IntegrationTestUtilities/FileUploadUtilities.cs]: Minor: If this is an async function now, can you add an async suffix here.

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/IntegrationTestUtilities/FileUploadUtilities.cs [hunk 4]
```diff
@@ -308,7 +308,7 @@ private static string CreateContainerWithPolicySASIfNotExist(string account, str
             return container_url;
         }
 
-        private static void CreateDefaultBlobContainerAndSASIfNeededReturn(List<IFileStagingProvider> filesToStage, SequentialFileStagingArtifact seqArtifact)
+        private static async Task CreateDefaultBlobContainerAndSASIfNeededReturn(List<IFileStagingProvider> filesToStage, SequentialFileStagingArtifact seqArtifact)
```
Comment (matthchr) [inline sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/IntegrationTestUtilities/FileUploadUtilities.cs]: Same, async suffix

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/IntegrationTestUtilities/SynchronizationContextHelper.cs [hunk 5]
```diff
@@ -17,7 +16,7 @@ private static void SetSynchronizationContext()
             //Note that XUnit sets the sync context and we want to make sure not to pollute that - this means that
             //we cannot use the Assert.<> functions which take a lamdba and do an await inside XUnit code because it 
             //will cause a deadlock...
-            DispatcherSynchronizationContext synchronizationContext = new DispatcherSynchronizationContext();
```
Comment (matthchr) [inline sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/IntegrationTestUtilities/SynchronizationContextHelper.cs]: This was actually intentional -- can you IFDEF this and make it this way in netfx, and take your change in netcore only (for now at least)?

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/Microsoft.Azure.Batch.Integration.Tests.csproj [hunk 6]
```diff
@@ -67,17 +53,14 @@
...
-    <ProjectReference Include="..\TestCommon\IntegrationTestCommon\IntegrationTestCommon.csproj" />
+    <ProjectReference Include="..\..\src\Microsoft.Azure.Batch.csproj" />
+    <ProjectReference Include="..\TestCommon\Microsoft.Azure.Batch.Common.Tests.csproj" />
+    <ProjectReference Include="..\IntegrationTestCommon\Microsoft.Azure.Batch.IntegrationTestCommon.Tests.csproj" />
   </ItemGroup>
   
   <ItemGroup>
     <Reference Include="System.Net" />
     <Reference Include="System.Net.Http" />
-    <Reference Include="System.Windows.Forms" />
```
Comment (matthchr) [inline sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/Microsoft.Azure.Batch.Integration.Tests.csproj]: Keep these in netfx/not in netcore.

File: sdk/batch/Microsoft.Azure.Batch/tests/ProtocolTests/IntegrationJobApplicationPackageReferencesTests.cs [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+            this.output = output;
+
+            IntegrationTestCommon.EnableAutoStorageAsync().Wait();
+
+            ApplicationPackageCommon.UploadTestApplicationPackageIfNotAlreadyUploadedAsync(AppPackageName, Version).Wait();
+            ApplicationPackageCommon.UpdateApplicationPackageAsync(AppPackageName, Version, "My First App", false).Wait();
+        }
+
+        //[Fact]
+        //[Trait(TestTraits.Duration.TraitName, TestTraits.Duration.Values.MediumDuration)]
```
Comment (matthchr) [inline sdk/batch/Microsoft.Azure.Batch/tests/ProtocolTests/IntegrationJobApplicationPackageReferencesTests.cs]: Was this stuff commented out before you moved it?

Comment (matthchr) [reply sdk/batch/Microsoft.Azure.Batch/tests/ProtocolTests/IntegrationJobApplicationPackageReferencesTests.cs]: Looks like it was

File: sdk/batch/Microsoft.Azure.Batch/tests/ProtocolTests/IntegrationJobApplicationPackageReferencesTests.cs [hunk 8]
```diff
@@ -0,0 +1,189 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (matthchr) [inline sdk/batch/Microsoft.Azure.Batch/tests/ProtocolTests/IntegrationJobApplicationPackageReferencesTests.cs]: Honestly if it's easier you can just delete this entire ProtocolTests project. It's all live tests though so if you don't you should flag it as such.

If you don't delete it I will shortly.

**Review by matthchr — CHANGES_REQUESTED** [review]
Took a look, it looks mostly ok, had a few comments.

**Comment by chidozieononiwu** [comment]
@Azure/azure-sdk-eng

File: eng/.docsettings.yml [hunk 9]
```diff
@@ -61,6 +61,8 @@ known_content_issues:
   - ['sdk/storage/Azure.Storage.Common/swagger/Generator/readme.md','azure-sdk-tools/issues/42']
   - ['sdk/cognitiveservices/Language.TextAnalytics/src/Readme.md','#5499']
   - ['sdk/cognitiveservices/Personalizer/src/Readme.md','#5499']
+  - ['sdk/batch/Microsoft.Azure.Batch/README.md','#5499']
```
[the remark this answers is not in the document]
Comment (chidozieononiwu) [reply eng/.docsettings.yml]: There is a Doc Warden script that verifies ReadMes' and enforces a certain set of minimum requirements. This line is added to skip that check while referencing the issue that describes it. So basically the ReadMe will have to be updated before this line can be removed else the build will break in CI. In this case the error would be 

> 1 readme has at least one missing required section.
    \sdk\batch\Microsoft.Azure.Batch\README.md is missing headers with patterns:
    * Azure .+ client library for .NET
    * ^Getting started$
    * ^Key concepts$
    * ^Examples$
    * ^Troubleshooting$
    * ^Next steps$
    * ^Contributing$
    For a rundown on what you need to do to resolve this breaking issue ASAP, check out 
    aka.ms/azure-sdk-analyze-failed

File: sdk/batch/Microsoft.Azure.Batch/src/Microsoft.Azure.Batch.csproj [hunk 10]
```diff
@@ -0,0 +1,56 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <Description>This client library provides access to the Microsoft Azure Batch service.</Description>
+    <VersionPrefix>10.0.0</VersionPrefix>
```
[the remark this answers is not in the document]
Comment (chidozieononiwu) [reply sdk/batch/Microsoft.Azure.Batch/src/Microsoft.Azure.Batch.csproj]: I did get the latest changes. I think I used the Assembly Version from [AssemblyAttributes.cs](https://github.com/Azure/azure-sdk-for-net/blob/2483720ad6ac1208e2044cd46490602353a9001b/src/SDKs/Batch/DataPlane/Azure.Batch/AssemblyAttributes.cs#L12) Should'nt they be the same?

File: sdk/batch/Microsoft.Azure.Batch/src/Microsoft.Azure.Batch.csproj [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+    <ProjectReference Include="..\..\Tools\ConfigureAwaitAnalyzer\ConfigureAwaitAnalyzer\ConfigureAwaitAnalyzer.csproj">
+      <ReferenceOutputAssembly>false</ReferenceOutputAssembly>
+      <Private>False</Private>
+    </ProjectReference>
+    <Analyzer Include="..\..\Tools\ConfigureAwaitAnalyzer\ConfigureAwaitAnalyzer\bin\Debug\netstandard1.4\ConfigureAwaitAnalyzer.dll" />
+    -->
+  </ItemGroup>
+
+  <PropertyGroup>
+    <AzureApiTag>Batch_2019-04-01;</AzureApiTag>
```
[the remark this answers is not in the document]
Comment (chidozieononiwu) [reply sdk/batch/Microsoft.Azure.Batch/src/Microsoft.Azure.Batch.csproj]: That's from [AzSdk.RP.props](https://github.com/Azure/azure-sdk-for-net/blob/2483720ad6ac1208e2044cd46490602353a9001b/src/SDKs/Batch/DataPlane/AzSdk.RP.props#L4) I just moved it into the batch file.

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTestCommon/CertificateBuilder.cs (hunk 1 above)
Comment (chidozieononiwu) [reply sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTestCommon/CertificateBuilder.cs]: For this class (CertificateBuilder.cs) the certificatebuilder was implemented with [Mono.Security Nuget](https://github.com/Azure/azure-sdk-for-net/blob/2483720ad6ac1208e2044cd46490602353a9001b/src/SDKs/Batch/DataPlane/TestCommon/IntegrationTestCommon/CertificateBuilder.cs#L9) which doesn't work with .NetCore and has been unlisted from Nuget.org. Here I am generating the self signed certificate with BountyCastle libraries. I really need you to take a look at it to be sure it works at it should.

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTestCommon/IntegrationTestCommon.cs [hunk 12]
```diff
@@ -100,8 +99,11 @@ private static CertificateCredentials GetBatchTestTenantCloudCredentials()
                     store.Close();
                 }
             }
-
+#if FullNetFx
             return new CertificateCredentials(certificate2);
```
Comment (chidozieononiwu) [inline sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTestCommon/IntegrationTestCommon.cs]: CertificateCredentials does not work with .NetCore2.1 can you suggest how to get around this?

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/ComputeNodeIntegrationTests.cs (hunk 2 above)
Comment (chidozieononiwu) [reply sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/ComputeNodeIntegrationTests.cs]: Am not sure what you mean, but I did make use of the continuation token in the implementation of the extension method.[here](https://github.com/Azure/azure-sdk-for-net/pull/6362/files#diff-b2ca8962af117800d7407f8b9ac85c57)

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/IntegrationTestUtilities/FileUploadUtilities.cs (hunk 3 above)
Comment (chidozieononiwu) [reply sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/IntegrationTestUtilities/FileUploadUtilities.cs]: Ok, will do.

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/IntegrationTestUtilities/FileUploadUtilities.cs (hunk 4 above)
Comment (chidozieononiwu) [reply sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/IntegrationTestUtilities/FileUploadUtilities.cs]: Ok, will do.

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/IntegrationTestUtilities/SynchronizationContextHelper.cs (hunk 5 above)
Comment (chidozieononiwu) [reply sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/IntegrationTestUtilities/SynchronizationContextHelper.cs]: Ok, will do.

File: sdk/batch/Microsoft.Azure.Batch/tests/ProtocolTests/IntegrationJobApplicationPackageReferencesTests.cs (hunk 8 above)
Comment (chidozieononiwu) [reply sdk/batch/Microsoft.Azure.Batch/tests/ProtocolTests/IntegrationJobApplicationPackageReferencesTests.cs]: Ok

File: sdk/batch/Microsoft.Azure.Batch/tests/ProtocolTests/IntegrationJobApplicationPackageReferencesTests.cs (hunk 7 above)
Comment (chidozieononiwu) [reply sdk/batch/Microsoft.Azure.Batch/tests/ProtocolTests/IntegrationJobApplicationPackageReferencesTests.cs]: No, it wasn't commented out earlier.
I did comment it out. This line
`var client = new BatchServiceClient(new Uri(Url), new BatchSharedKeyCredential(AccountName, AccountKey))` was an error due to wrong arguements.

>Error	CS1503	Argument 1: cannot convert from 'System.Uri' to 'Microsoft.Rest.ServiceClientCredentials'	Microsoft.Azure.Batch.Protocol.Tests (net461), Microsoft.Azure.Batch.Protocol.Tests (netcoreapp2.1)	

> Error	CS1503	Argument 2: cannot convert from 'Microsoft.Azure.Batch.Protocol.BatchSharedKeyCredential' to 'System.Net.Http.DelegatingHandler'	Microsoft.Azure.Batch.Protocol.Tests (net461), Microsoft.Azure.Batch.Protocol.Tests (netcoreapp2.1)

File: sdk/batch/Microsoft.Azure.Batch/src/Microsoft.Azure.Batch.csproj (hunk 10 above)
[the remark this answers is not in the document]
Comment (matthchr) [reply sdk/batch/Microsoft.Azure.Batch/src/Microsoft.Azure.Batch.csproj]: They probably _should_ be the same. If they weren't though use the later one.

File: sdk/batch/Microsoft.Azure.Batch/src/Microsoft.Azure.Batch.csproj (hunk 11 above)
[the remark this answers is not in the document]
Comment (matthchr) [reply sdk/batch/Microsoft.Azure.Batch/src/Microsoft.Azure.Batch.csproj]: Why is the data plane SDK inheriting anything from AzSdk.RP.props? It's not an RP?

This value is wrong so I don't think we should have it here - can we just remove it?

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTestCommon/CertificateBuilder.cs (hunk 1 above)
Comment (matthchr) [reply sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTestCommon/CertificateBuilder.cs]: Yes, I looked and this looked okay to me. Also, I ran tests which use this (in IntegrationTests) and they passed. So you're good here.

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/ComputeNodeIntegrationTests.cs (hunk 2 above)
Comment (matthchr) [reply sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/ComputeNodeIntegrationTests.cs]: I mean that the Azure Storage SDK usually does this for you. The ListBlobs method we were calling previously already did this - I'm wondering why we can't just use an existing ListBlobs from the storage SDK itself rather than write our own extension

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTestCommon/IntegrationTestCommon.cs (hunk 12 above)
Comment (matthchr) [reply sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTestCommon/IntegrationTestCommon.cs]: Do you know what the alternative is for CertificateCredentials in netcore? CertificateCredentials AFAIK comes from the Azure SDK so presumably there's some replacement for it?

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/ComputeNodeIntegrationTests.cs (hunk 2 above)
Comment (chidozieononiwu) [reply sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/ComputeNodeIntegrationTests.cs]: The ListBlobs method is not available in .netcore2.1

File: sdk/batch/Microsoft.Azure.Batch/Tools/CodeGenerationLibrary/CodeGenerationLibrary.csproj [hunk 13]
```diff
@@ -1,18 +1,17 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <PackageId>CodeGenerationLibrary</PackageId>
     <Description>Azure Batch object model code generation library</Description>
     <!--After fixing all warnings, enable this option-->
     <TreatWarningsAsErrors>false</TreatWarningsAsErrors>
-    <AssemblyName>CodeGenerationLibrary</AssemblyName>
+    <AssemblyTitle>CodeGenerationLibrary</AssemblyTitle>
```
Comment (weshaggard) [inline sdk/batch/Microsoft.Azure.Batch/Tools/CodeGenerationLibrary/CodeGenerationLibrary.csproj]: Why this change? I wouldn't expect you need AssemblyName or AssemblyTitle.

File: sdk/batch/Microsoft.Azure.Batch/Tools/CodeGenerationLibrary/CodeGenerationLibrary.csproj [hunk 14]
```diff
@@ -1,18 +1,17 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <PackageId>CodeGenerationLibrary</PackageId>
     <Description>Azure Batch object model code generation library</Description>
     <!--After fixing all warnings, enable this option-->
     <TreatWarningsAsErrors>false</TreatWarningsAsErrors>
```
Comment (weshaggard) [inline sdk/batch/Microsoft.Azure.Batch/Tools/CodeGenerationLibrary/CodeGenerationLibrary.csproj]: Looks like this is set to false here and then to true a few lines down.

File: sdk/batch/Microsoft.Azure.Batch/Tools/ConfigureAwaitAnalyzer/ConfigureAwaitAnalyzer.csproj [hunk 15]
```diff
@@ -1,14 +1,15 @@
...
-    <PackageId>ConfigureAwaitAnalyzer</PackageId>
     <Description>Azure Batch Roslyn analyzer for checking configureawait</Description>
     <!--After fixing all warnings, enable this option-->
     <TreatWarningsAsErrors>false</TreatWarningsAsErrors>
     <DefineConstants>$(DefineConstants);TESTSIGN</DefineConstants>
-    <AssemblyName>ConfigureAwaitAnalyzer</AssemblyName>
+    <AssemblyTitle>ConfigureAwaitAnalyzer</AssemblyTitle>
     <SignAssembly>true</SignAssembly>
     <DelaySign>true</DelaySign>
+    <AssemblyOriginatorKeyFile>$(RepoEngPath)\MSSharedLibKey.snk</AssemblyOriginatorKeyFile>
```
Comment (weshaggard) [inline sdk/batch/Microsoft.Azure.Batch/Tools/ConfigureAwaitAnalyzer/ConfigureAwaitAnalyzer.csproj]: This shouldn't be needed.

File: sdk/batch/Microsoft.Azure.Batch/Tools/ConfigureAwaitAnalyzer/ConfigureAwaitAnalyzer.csproj [hunk 16]
```diff
@@ -1,14 +1,15 @@
...
     <RestorePackagesPath>$(LibraryNugetPackageFolder)</RestorePackagesPath>
+    <GenerateAssemblyInformationalVersionAttribute>false</GenerateAssemblyInformationalVersionAttribute>
```
Comment (weshaggard) [inline sdk/batch/Microsoft.Azure.Batch/Tools/ConfigureAwaitAnalyzer/ConfigureAwaitAnalyzer.csproj]: Why are you disabling the assembly information attribute?

File: sdk/batch/Microsoft.Azure.Batch/Tools/ConfigureAwaitAnalyzerTests/ConfigureAwaitAnalyzer.Tests.csproj [hunk 17]
```diff
@@ -10,21 +8,20 @@
     <SkipBuildingTestProject>false</SkipBuildingTestProject>
     <SignAssembly>true</SignAssembly>
     <DelaySign>true</DelaySign>
-    <AssemblyOriginatorKeyFile>$(LibraryToolsFolder)\MSSharedLibKey.snk</AssemblyOriginatorKeyFile>
+    <AssemblyOriginatorKeyFile>$(RepoEngPath)\MSSharedLibKey.snk</AssemblyOriginatorKeyFile>
```
Comment (weshaggard) [inline sdk/batch/Microsoft.Azure.Batch/Tools/ConfigureAwaitAnalyzerTests/ConfigureAwaitAnalyzer.Tests.csproj]: Shouldn't be necessary.

File: sdk/batch/Microsoft.Azure.Batch/src/AssemblyAttributes.cs [hunk 18]
```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+
+#if CODESIGN || TESTSIGN
```
Comment (weshaggard) [inline sdk/batch/Microsoft.Azure.Batch/src/AssemblyAttributes.cs]: We always sign now so this shouldn't be needed.

File: sdk/batch/Microsoft.Azure.Batch/Tools/CodeGenerationLibrary/CodeGenerationLibrary.csproj (hunk 13 above)
Comment (chidozieononiwu) [reply sdk/batch/Microsoft.Azure.Batch/Tools/CodeGenerationLibrary/CodeGenerationLibrary.csproj]: Why would it not be needed? Is it because it is not a shipped package?

File: sdk/batch/Microsoft.Azure.Batch/Tools/CodeGenerationLibrary/CodeGenerationLibrary.csproj [hunk 19]
```diff
@@ -1,18 +1,17 @@
...
-    <AssemblyName>CodeGenerationLibrary</AssemblyName>
+    <AssemblyTitle>CodeGenerationLibrary</AssemblyTitle>
     <RestorePackagesPath>$(LibraryNugetPackageFolder)</RestorePackagesPath>
```
Comment (chidozieononiwu) [inline sdk/batch/Microsoft.Azure.Batch/Tools/CodeGenerationLibrary/CodeGenerationLibrary.csproj]: @weshaggard Does this apply in for dataplane?

File: sdk/batch/Microsoft.Azure.Batch/Tools/CodeGenerationLibrary/CodeGenerationLibrary.csproj (hunk 13 above)
Comment (weshaggard) [reply sdk/batch/Microsoft.Azure.Batch/Tools/CodeGenerationLibrary/CodeGenerationLibrary.csproj]: It isn't needed for the same reason AssemblyName isn't needed because it is redundant and defaults from the AssemblyName which defaults from the Project Name, https://github.com/dotnet/sdk/blob/2eb6c546931b5bcb92cd3128b93932a980553ea1/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DefaultAssemblyInfo.targets#L23.

File: sdk/batch/Microsoft.Azure.Batch/Tools/CodeGenerationLibrary/CodeGenerationLibrary.csproj (hunk 19 above)
Comment (weshaggard) [reply sdk/batch/Microsoft.Azure.Batch/Tools/CodeGenerationLibrary/CodeGenerationLibrary.csproj]: I don't believe RestorePackagesPath applies.

Comment (weshaggard) [reply sdk/batch/Microsoft.Azure.Batch/Tools/CodeGenerationLibrary/CodeGenerationLibrary.csproj]: That said however I'm not entirely sure what it is being used for, it might be getting used by something other then nuget.

File: sdk/batch/Microsoft.Azure.Batch/Tools/CodeGenerationLibrary/CodeGenerationLibrary.csproj (hunk 13 above)
Comment (chidozieononiwu) [reply sdk/batch/Microsoft.Azure.Batch/Tools/CodeGenerationLibrary/CodeGenerationLibrary.csproj]: Thanks

File: sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTestCommon/IntegrationTestCommon.cs (hunk 12 above)
Comment (chidozieononiwu) [reply sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTestCommon/IntegrationTestCommon.cs]: I haven't found an alternative foe that.

**Comment by weshaggard** [comment]
Looks like there are some warnings about test results because some of the test projects don't actually contain tests:

https://dev.azure.com/azure-sdk/public/_build/results?buildId=38551&view=logs - PublishTestResults step

Windows:
```
No Result Found to Publish 'D:\a\1\s\sdk\batch\Microsoft.Azure.Batch\tests\IntegrationTestCommon\TestResults\VssAdministrator_fv-az639_2019-05-25_01_57_42_499.trx'.
No Result Found to Publish 'D:\a\1\s\sdk\batch\Microsoft.Azure.Batch\tests\IntegrationTestCommon\TestResults\VssAdministrator_fv-az639_2019-05-25_01_57_57_715.trx'.
No Result Found to Publish 'D:\a\1\s\sdk\batch\Microsoft.Azure.Batch\tests\ProtocolTests\TestResults\VssAdministrator_fv-az639_2019-05-25_01_58_10_982.trx'.
No Result Found to Publish 'D:\a\1\s\sdk\batch\Microsoft.Azure.Batch\tests\ProtocolTests\TestResults\VssAdministrator_fv-az639_2019-05-25_01_58_14_742.trx'.
No Result Found to Publish 'D:\a\1\s\sdk\batch\Microsoft.Azure.Batch\tests\TestCommon\TestResults\VssAdministrator_fv-az639_2019-05-25_01_58_10_550.trx'.
No Result Found to Publish 'D:\a\1\s\sdk\batch\Microsoft.Azure.Batch\tests\TestCommon\TestResults\VssAdministrator_fv-az639_2019-05-25_01_58_12_335.trx'.
```

The linux and Mac give this warning instead:

```
##[warning]Invalid results file. Make sure the result format of the file '/home/vsts/work/1/s/sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTestCommon/TestResults/_fv-az644_2019-05-25_01_56_30_382.trx' matches 'VSTest' test results format.
##[warning]Invalid results file. Make sure the result format of the file '/home/vsts/work/1/s/sdk/batch/Microsoft.Azure.Batch/tests/ProtocolTests/TestResults/_fv-az644_2019-05-25_01_56_41_929.trx' matches 'VSTest' test results format.
##[warning]Invalid results file. Make sure the result format of the file '/home/vsts/work/1/s/sdk/batch/Microsoft.Azure.Batch/tests/TestCommon/TestResults/_fv-az644_2019-05-25_01_56_33_026.trx' matches 'VSTest' test results format.
```

However I think they stem from the same issue which is there are not test results for those projects.  We should either rename the projects or set IsTestProject=false (similar to     
    <IsTestProject Condition="'$(IsTestSupportProject)' == 'true'">false</IsTestProject>) in the project to get them not to run tests and report no results.

**Comment by chidozieononiwu** [comment]
> Looks like there are some warnings about test results because some of the test projects don't actually contain tests:
> 
> https://dev.azure.com/azure-sdk/public/_build/results?buildId=38551&view=logs - PublishTestResults step
> 
> Windows:
> 
> ```
> No Result Found to Publish 'D:\a\1\s\sdk\batch\Microsoft.Azure.Batch\tests\IntegrationTestCommon\TestResults\VssAdministrator_fv-az639_2019-05-25_01_57_42_499.trx'.
> No Result Found to Publish 'D:\a\1\s\sdk\batch\Microsoft.Azure.Batch\tests\IntegrationTestCommon\TestResults\VssAdministrator_fv-az639_2019-05-25_01_57_57_715.trx'.
> No Result Found to Publish 'D:\a\1\s\sdk\batch\Microsoft.Azure.Batch\tests\ProtocolTests\TestResults\VssAdministrator_fv-az639_2019-05-25_01_58_10_982.trx'.
> No Result Found to Publish 'D:\a\1\s\sdk\batch\Microsoft.Azure.Batch\tests\ProtocolTests\TestResults\VssAdministrator_fv-az639_2019-05-25_01_58_14_742.trx'.
> No Result Found to Publish 'D:\a\1\s\sdk\batch\Microsoft.Azure.Batch\tests\TestCommon\TestResults\VssAdministrator_fv-az639_2019-05-25_01_58_10_550.trx'.
> No Result Found to Publish 'D:\a\1\s\sdk\batch\Microsoft.Azure.Batch\tests\TestCommon\TestResults\VssAdministrator_fv-az639_2019-05-25_01_58_12_335.trx'.
> ```
> 
> The linux and Mac give this warning instead:
> 
> ```
> ##[warning]Invalid results file. Make sure the result format of the file '/home/vsts/work/1/s/sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTestCommon/TestResults/_fv-az644_2019-05-25_01_56_30_382.trx' matches 'VSTest' test results format.
> ##[warning]Invalid results file. Make sure the result format of the file '/home/vsts/work/1/s/sdk/batch/Microsoft.Azure.Batch/tests/ProtocolTests/TestResults/_fv-az644_2019-05-25_01_56_41_929.trx' matches 'VSTest' test results format.
> ##[warning]Invalid results file. Make sure the result format of the file '/home/vsts/work/1/s/sdk/batch/Microsoft.Azure.Batch/tests/TestCommon/TestResults/_fv-az644_2019-05-25_01_56_33_026.trx' matches 'VSTest' test results format.
> ```
> 
> However I think they stem from the same issue which is there are not test results for those projects. We should either rename the projects or set IsTestProject=false (similar to https://github.com/Azure/azure-sdk-for-net/blob/master/eng/Directory.Build.Data.targets#L31) in the project to get them not to run tests and report no results.

I have fixed the be renaming the common tests projects. I also deleted the protocol test as @matthchr requested. I guess this is ready to merge now.

**Review by weshaggard — APPROVED** [review]

**Review by matthchr — APPROVED** [review]
Looks good now
